### PR TITLE
[Feature][Branch][WIP] add CalendarCarousel

### PR DESCRIPTION
**Pull request by kty1965**
## Description

Create CalendarCarousel shared component.
I used other libraries.

- [date-fns](https://date-fns.org/)
- [typescript-fsa](https://github.com/aikoven/typescript-fsa)

This pull request produce 2 components
- Calendar
- CalendarCarousel

## Related Issues

**Cannot find Issue**

## Tests

**Need to Test**

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui-native/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [ ] I signed the [CLA].
- [ ] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [ ] Run `yarn lint`
- [ ] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by kty1965** [comment]
Added 2 libraries in `src/components/shared/CalendarCarousel/package.json`.
But Circleci doesn't install sub-modules.

File: package.json [hunk 1]
```diff
@@ -31,7 +31,9 @@
   },
   "homepage": "https://github.com/dooboolab/dooboo-ui-native#readme",
   "dependencies": {
-    "dooboolab-welcome": "^1.1.1"
+    "dooboolab-welcome": "^1.1.1",
+    "styled-components": "^4.4.0",
```
Comment (hyochan) [inline package.json]: Please remove this.

File: package.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "codecov": "codecov",
     "git:log": "git log --decorate --oneline --all --graph",
     "watch": "lerna exec -- yarn && lerna run --parallel watch",
-    "pre": "lerna exec -- yarn && lerna run build"
+    "pre": "lerna exec --concurrency 1 -- yarn && lerna run build"
```
Comment (hyochan) [inline package.json:19]: 👍

**Comment by hyochan** [comment]
@kty1965 Thanks for your effort in contributing to this awesome component. However, I think we should bring this up again when we are ready. We are currently planning for massive migration! I'll remember this `PR` and bring it back!

**Pull request closed without merging**